api/envoy/extensions/load_balancing_policies/common/v3/common.proto

```diff
@@ -42,8 +43,22 @@ message LocalityLbConfig {
     // failing service.
     bool fail_traffic_on_panic = 3;
 
-    // If set to true, Envoy will force LocalityDirect routing if a local locality exists.
-    bool force_locality_direct_routing = 4;
+    // If set to true, Envoy will always route traffic to the local zone regardless of the
+    // upstream zone structure. In Envoy's default configuration, traffic is distributed proportionally
+    // across all upstream hosts while trying to maximize local routing when possible. The approach
+    // with force_local_zone aims to be more predictable and if there are upstream hosts in the local
+    // zone, they will receive all traffic.
+    // * :ref:`runtime values <config_cluster_manager_cluster_runtime_zone_routing>`.
+    // * :ref:`Zone aware routing support <arch_overview_load_balancing_zone_aware_routing>`.
+    bool force_local_zone = 4;
+
+    // Configures the minimum number of upstream hosts in the local zone required when force_local_zone
+    // is enabled. If the number of upstream hosts in the local zone is less than the specified value,
+    // Envoy will fall back to the default proportional-based distribution across localities.
+    // If not specified, the default is 1.
+    // * :ref:`runtime values <config_cluster_manager_cluster_runtime_zone_routing>`.
+    // * :ref:`Zone aware routing support <arch_overview_load_balancing_zone_aware_routing>`.
+    google.protobuf.UInt32Value force_local_zone_min_size = 5;
```

Comment:
Change overall looks good to me. Just having a practical thought, how users would pick up  `force_local_zone_min_size` number per cluster/service, e.g. the good number depends on service throughput, amount of traffic, scaling events etc. Would it be more practical to instead express this value as percentage? (e.g. at least 10% of total endpoints are available in local zone). cc @tonya11en @wbpcode

Reply:
The intention of `force_local_zone` is to disregard the structure of other localities so I'm not sure that'd be a good fit here. The default logic mostly covers that already by ensuring equal distrubtion as a result of calculating the zone structure.

My intention with this field was to offer a lever to owners of the upstream cluster so they can indicate a minimum for the zone similar to how `min_cluster_size` works in the default logic.